src/editor/CustomEditor.ts

```diff
@@ -98,10 +101,102 @@ const createCustomEditor = () => {
     }
   };
 
+  /**
+   * This function returns true if the currently selected block is a void orphan,
+   * that is, an empty paragraph at the very beginning or end of the document
+   * before or after a void block and its spacers.
+   */
+  const isVoidOrphan = () => {
+    if (!editor.selection || !Range.isCollapsed(editor.selection)) {
+      return false;
+    }
+    if (
+      !Editor.isStart(editor, editor.selection.anchor, []) &&
+      !Editor.isEnd(editor, editor.selection.anchor, [])
+    ) {
+      return false;
+    }
+    if (Editor.string(editor, editor.selection.anchor.path) !== "") {
+      return false;
+    }
+    return true;
+  };
+
+  const deleteForwardCustom: typeof deleteForward = (unit) => {
+    if (
+      editor.selection &&
+      Range.isCollapsed(editor.selection) &&
```

Comment:
This particular `editor.selection && Range.isCollapsed(editor.selection)` pattern appears a few times, do we want to wrap this into a function?

Reply:
Good catch: https://github.com/UWmathNEWS/slugline-web/pull/84/commits/b94d41ba74a36664f8296c6892f5282b6108ccac